cpp/src/parquet/file_reader.cc

```diff
@@ -266,13 +266,17 @@ class SerializedRowGroup : public RowGroupReader::Contents {
     ARROW_DCHECK_NE(meta_decryptor, nullptr);
     ARROW_DCHECK_NE(data_decryptor, nullptr);
 
-    constexpr auto kEncryptedRowGroupsLimit = 32767;
-    if (i > kEncryptedRowGroupsLimit) {
+    constexpr auto kEncryptedOrdinalLimit = 32767;
+    if (ARROW_PREDICT_FALSE(row_group_ordinal_ > kEncryptedOrdinalLimit)) {
```

Comment:
I am wondering if this is the best place to enforce this. Is it possible to check these while creating the `FileMetaData` if encrypted?

Reply:
It seems that if we don't request reading data from `row_group_ordinal > 32767`  and `column_ordinal > 32767`, we will even not notice that it is a malformed encrypted file.

Reply:
Do you mean when contructing `ColumnChunkMetaData::ColumnChunkMetaDataImpl` ? What if the `ENCRYPTION_WITH_COLUMN_KEY` is not set but this is being encrypted?